rust-runtime/aws-smithy-http-server/src/extension.rs

```diff
@@ -69,8 +69,11 @@ impl ResponseExtensions {
 pub struct ExtensionModeledError(&'static str);
 impl_extension_new_and_deref!(ExtensionModeledError);
 
+// TODO This should probably be renamed to `ExtensionException`, since we now surface exceptions.
+// Although I prefer to put `Extension` at the end for these, like we do for the types in
+// `rejection.rs`, so, `ExceptionExtension`.
```

Comment:
I agree about the naming, `ExceptionExtension` works for me here.

Reply:
I like 1) and 2) from your proposal.

Regarding 3), shouldn't we keep a way to signal the user than something bad happened at the framework level? In my mind a `ModeledError` is something defined in the user Smithy model, while the other can be stuff like deserialization errors and so on..

Reply:
We will rename `ResponseExtension` to `OperationExtension` and `ExtensionModeledError` to `ModeledErrorExtension`. We will add `RuntimeErrorExtension`. We're unsure whether we should panic or not when a request extension is not found. For now, let's keep returning a 500 response and include a `RequestExtensionNotFoundExtension` in the response.